### ISSUE TITLE
Can't run against simulator from inside Xcode, ld: symbol(s) not found for architecture i386

### ISSUE THREAD

**Issue by xster** [flutter/flutter#17749]
Open a simulator, then `flutter run`. Gallery runs ok. Then open the Gallery Xcode project in Xcode and run against the same simulator. 

```
Ld /Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Intermediates.noindex/Runner.build/Release-iphonesimulator/Runner.build/Objects-normal/i386/Runner normal i386
    cd /Users/xiao/Development/flutter/examples/flutter_gallery/ios
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch i386 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.3.sdk -L/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator -L/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator/Reachability -L/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator/connectivity -L/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator/device_info -L/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator/url_launcher -L/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator/video_player -L/Users/xiao/Development/flutter/examples/flutter_gallery/ios/Flutter -F/Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Products/Release-iphonesimulator -F/Users/xiao/Development/flutter/examples/flutter_gallery/ios/Pods/../../../../bin/cache/artifacts/engine/ios -F/Users/xiao/Development/flutter/examples/flutter_gallery/ios/Flutter -filelist /Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Intermediates.noindex/Runner.build/Release-iphonesimulator/Runner.build/Objects-normal/i386/Runner.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=8.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Intermediates.noindex/Runner.build/Release-iphonesimulator/Runner.build/Objects-normal/i386/Runner_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -ObjC -lReachability -lconnectivity -ldevice_info -lurl_launcher -lvideo_player -framework Flutter -framework SystemConfiguration -ObjC -lReachability -lconnectivity -ldevice_info -lurl_launcher -lvideo_player -framework Flutter -framework SystemConfiguration -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Intermediates.noindex/Runner.build/Release-iphonesimulator/Runner.build/Runner.app-Simulated.xcent -framework Flutter -framework App -lPods-Runner -Xlinker -dependency_info -Xlinker /Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Intermediates.noindex/Runner.build/Release-iphonesimulator/Runner.build/Objects-normal/i386/Runner_dependency_info.dat -o /Users/xiao/Library/Developer/Xcode/DerivedData/Runner-dokcgemxpxmqtfbsjxggqrmijoqp/Build/Intermediates.noindex/Runner.build/Release-iphonesimulator/Runner.build/Objects-normal/i386/Runner

ld: warning: ignoring file /Users/xiao/Development/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter, missing required architecture i386 in file /Users/xiao/Development/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter (3 slices)
Undefined symbols for architecture i386:
  "_OBJC_CLASS_$_FlutterError", referenced from:
      objc-class-ref in liburl_launcher.a(UrlLauncherPlugin.o)
      objc-class-ref in libvideo_player.a(VideoPlayerPlugin.o)
  "_OBJC_CLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_CLASS_$_AppDelegate in AppDelegate.o
  "_FlutterMethodNotImplemented", referenced from:
      -[FLTConnectivityPlugin handleMethodCall:result:] in libconnectivity.a(ConnectivityPlugin.o)
      -[FLTDeviceInfoPlugin handleMethodCall:result:] in libdevice_info.a(DeviceInfoPlugin.o)
      -[FLTUrlLauncherPlugin handleMethodCall:result:] in liburl_launcher.a(UrlLauncherPlugin.o)
      -[FLTVideoPlayerPlugin handleMethodCall:result:] in libvideo_player.a(VideoPlayerPlugin.o)
  "_OBJC_METACLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_METACLASS_$_AppDelegate in AppDelegate.o
  "_OBJC_CLASS_$_FlutterEventChannel", referenced from:
      objc-class-ref in libconnectivity.a(ConnectivityPlugin.o)
      objc-class-ref in libvideo_player.a(VideoPlayerPlugin.o)
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in libconnectivity.a(ConnectivityPlugin.o)
      objc-class-ref in libdevice_info.a(DeviceInfoPlugin.o)
      objc-class-ref in liburl_launcher.a(UrlLauncherPlugin.o)
      objc-class-ref in libvideo_player.a(VideoPlayerPlugin.o)
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

cc @cbracken

**Comment by xster** [comment]
Ummm totally TIL https://stackoverflow.com/questions/28033635/why-does-the-ios-simulator-require-i386-and-x86-64-symbols-even-though-im-on-an. 

Wonder if I set up my Xcode incorrectly. Though I don't think I did anything special.

**Comment by MichaelShnaider** [comment]
@efortuna I've been experiencing the "architecture i386" as well, if a known fix already exists, it would be amazing if someone could point me in the right direction.

For me, to replicate the issue, I was following the tutorial:
https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps

and for iOS when I set the iOS Target Deployment to below 11, I get the "architecture i386" error and when I set the iOS Target Deployment to above 11, I don't get the error. 

I noticed the OP has "IPHONEOS_DEPLOYMENT_TARGET=8.0" in the error, which may be the cause, if the 8.0 is replaced with a "11.0" or above, then the error will probably go away.

However, for my use case I need to have the Target Deployment set below 11 for now :'(

**Comment by dilbert2** [comment]
i found this:
https://github.com/flutter/engine/wiki/ARMv7(iOS)-&-armeabi-v7a(Android)-build-with-Xcode10

**Comment by ryansong** [comment]
@dilbert2 
how to find the check commend
"your-flutter-engine-path/engine/src/build/mac/find_sdk.py", I can't find it in Flutter Path.

**Comment by vipinagrahari** [comment]
Is there any solution available for this problem?

**Comment by supermnemonic** [comment]
I have same issue when trying on simulator any iphone (iOS 10 and below). Any workaround for this?

Flutter (Channel master, v1.3.10-pre.16, on Mac OS X 10.14.3 18D109, locale en-ID)


    ld: warning: ignoring file /Users/xxxxxx/development/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter, missing required architecture i386 in file /Users/xxxxxx/development/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter (3 slices)
    Undefined symbols for architecture i386:
      "_OBJC_CLASS_$_FlutterStandardTypedData", referenced from:
          objc-class-ref in libsqflite.a(SqflitePlugin.o)
      "_OBJC_CLASS_$_FlutterStandardMessageCodec", referenced from:
          objc-class-ref in libgoogle_maps_flutter.a(GoogleMapController.o)
      "_OBJC_CLASS_$_FlutterAppDelegate", referenced from:
          _OBJC_CLASS_$_AppDelegate in AppDelegate.o
      "_FlutterMethodNotImplemented", referenced from:
          -[FLTFirebaseAdMobPlugin handleMethodCall:result:] in libfirebase_admob.a(FirebaseAdMobPlugin.o)
          -[FLTFirebaseAnalyticsPlugin handleMethodCall:result:] in libfirebase_analytics.a(FirebaseAnalyticsPlugin.o)
          -[FLTFirebaseCorePlugin handleMethodCall:result:] in libfirebase_core.a(FirebaseCorePlugin.o)
          -[FLTGoogleMapController onMethodCall:result:] in libgoogle_maps_flutter.a(GoogleMapController.o)
          ___47+[FLTPathProviderPlugin registerWithRegistrar:]_block_invoke in libpath_provider.a(PathProviderPlugin.o)
          ___52+[FLTSharedPreferencesPlugin registerWithRegistrar:]_block_invoke in libshared_preferences.a(SharedPreferencesPlugin.o)
      -[SqflitePlugin handleMethodCall:result:] in libsqflite.a(SqflitePlugin.o)
      ...
      "_OBJC_CLASS_$_FlutterError", referenced from:
      objc-class-ref in libfirebase_admob.a(FirebaseAdMobPlugin.o)
      objc-class-ref in libgoogle_maps_flutter.a(GoogleMapsPlugin.o)
      objc-class-ref in libsqflite.a(SqflitePlugin.o)
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in libfirebase_admob.a(FirebaseAdMobPlugin.o)
      objc-class-ref in libfirebase_analytics.a(FirebaseAnalyticsPlugin.o)
      objc-class-ref in libfirebase_core.a(FirebaseCorePlugin.o)
      objc-class-ref in libgoogle_maps_flutter.a(GoogleMapController.o)
      objc-class-ref in libpath_provider.a(PathProviderPlugin.o)
      objc-class-ref in libshared_preferences.a(SharedPreferencesPlugin.o)
      objc-class-ref in libsqflite.a(SqflitePlugin.o)
      ...
      "_OBJC_METACLASS_$_FlutterAppDelegate", referenced from:
      _OBJC_METACLASS_$_AppDelegate in AppDelegate.o
    ld: symbol(s) not found for architecture i386
    clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by jobpav1984** [comment]
you have to have to set ONLY_ACTIVE_ARCH = YES for debug builds. Write this to you podfile

`
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['SWIFT_VERSION'] = '4.2'
      config.build_settings['SUPPORTED_PLATFORMS'] = 'iphoneos iphonesimulator'
      if config.name == "Debug" || config.name == "Debug-alpha" || config.name == "Debug-live" || config.name == "Debug-development"
        config.build_settings['ONLY_ACTIVE_ARCH'] = 'YES'
      else
        config.build_settings['ONLY_ACTIVE_ARCH'] = 'NO'
      end
    end
  end
end
`

**Comment by PandaGeek1024** [comment]
I came across this issue when I set multiple flavors. It works totally fine with the original setup. I don't know why this is happening with multiple flavors. The only solution for me so far is to set Architecture for arm64 only under build setting. But I really do want to know why it works totally fine with original setup.

**Comment by NickNarayana** [comment]
I see below error trying to build the code. works only for one debug scheme but not for release/enterprise or any other...

ld: warning: ignoring file /Users/.../Project/flutter/.ios/Flutter/engine/Flutter.framework/Flutter, missing required architecture i386 in file /Users/.../Project/flutter/.ios/Flutter/engine/Flutter.framework/Flutter (3 slices)
Undefined symbols for architecture i386:
  "_FlutterMethodNotImplemented", referenced from:
      ___47+[FLTPathProviderPlugin registerWithRegistrar:]_block_invoke in PathProviderPlugin.o
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in PathProviderPlugin.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by worldofsites** [comment]
Same issue.

**Comment by searchy2** [comment]
Same issue here. Changing iOS Build Target to 11.0 fixed the issue for me too.

**Comment by nmfisher** [comment]
If anyone's experiencing "symbols not found for i386", in addition to checking your main project's build target, check if any plugin podspec files specify a deployment target:
```
s.ios.deployment_target = '8.0'
```

Increase this to a later version (11.0+), or even better, just delete the line altogether and let the project manage the deployment target. I'm not 100% sure, but it seems that Flutter drops its build targets to the lowest common denominator amongst all project/plugin deployment targets.

**Comment by mleonhard** [comment]
Updated link to wiki article with instructions for building 32-bit apps:
https://github.com/flutter/flutter/wiki/Supporting-legacy-platforms

I'm about to give up on supporting 32-bit devices, iPhone 5 and older.  Is there a reliable source of data about how many people use various iPhone models?

**Comment by nicowernli** [comment]
> I came across this issue when I set multiple flavors. It works totally fine with the original setup. I don't know why this is happening with multiple flavors. The only solution for me so far is to set Architecture for arm64 only under build setting. But I really do want to know why it works totally fine with original setup.
> 
> `ld: warning: ignoring file /Users/pengfeihuang/Developer/Flutter/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter, missing required architecture i386 in file /Users/pengfeihuang/Developer/Flutter/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter (3 slices) Undefined symbols for architecture i386: "_FlutterMethodNotImplemented", referenced from: ___47+[FLTPathProviderPlugin registerWithRegistrar:]_block_invoke in PathProviderPlugin.o "_OBJC_CLASS_$_FlutterMethodChannel", referenced from: objc-class-ref in PathProviderPlugin.o ld: symbol(s) not found for architecture i386 clang: error: linker command failed with exit code 1 (use -v to see invocation)`

how do you do this? Could you help me doing it? I'm also having this issue when implementing flutter flavors

**Comment by jmagman** [comment]
Flutter frameworks contain armv7, arm64, and x86_64 slices but NOT i386.  When you run in the simulator, Xcode silently maps armv7 to i386 simulator and arm64 to the x86_64 simulator.  `ONLY_ACTIVE_ARCH=YES` build setting builds only for the architecture of the device or simulator you are targeting.  Xcode docs:

> If enabled, only the active architecture is built. This setting will be ignored when building with a run destination which does not define a specific architecture, such as a 'Generic Device' run destination.

So if you are targeting a real recent iOS device, it will build arm64, and if you target a recent simulator, it will build x86_64.  `ONLY_ACTIVE_ARCH=YES` is the default for the Debug build configuration for new Xcode projects.

If you have `ONLY_ACTIVE_ARCH=NO` and target a real device, it will build armv7 and arm64 and will succeed.  If you target a simulator, it will try to build i386 and x86_64 and fail with this error because i386 is missing.  `ONLY_ACTIVE_ARCH=NO` is the default for Release build configurations for new Xcode projects because you need to publish all valid architectures to the App Store.

[Flutter is not supported in Release on the simulator.](https://github.com/flutter/flutter/issues/11754)

TL;DR As @jobpav1984 mentions, make sure you have `ONLY_ACTIVE_ARCH=YES` for any Debug build configurations.  You must have `ONLY_ACTIVE_ARCH=NO` for any build configurations you intend to publish to the App Store.

I recommend doing this in Xcode's build settings, NOT in your Podfile.

<img width="953" alt="Screen Shot 2020-01-10 at 11 07 52 AM" src="https://user-images.githubusercontent.com/682784/72179297-7558fb00-3399-11ea-9089-429cf2ac0473.png">

If anyone is hitting this error with `ONLY_ACTIVE_ARCH=YES` that would be interesting to know.

**Comment by csells** [comment]
@xster does the suggested work-around fix your issue?

**Comment by matibzurovski** [comment]
> If anyone is hitting this error with ONLY_ACTIVE_ARCH=YES that would be interesting to know.

I am having the same situation with `ONLY_ACTIVE_ARCH=YES` @jmagman. I've tried `beta` and `master` channel of Flutter and it failed in both of them.

**Comment by sgehrman** [comment]
same problem, can't find a solution yet.

**Comment by rubensdemelo** [comment]
same

**Comment by timeuser** [comment]
To fix this and get it building for Simulator I had to set `ONLY_ACTIVE_ARCH=YES` for the debug schemes Pods project also, in addition to the same settings the Runner project. It's working for me now.

<img width="1167" alt="ActiveArchitecture" src="https://user-images.githubusercontent.com/119743/79580266-636caf80-808e-11ea-8d97-6f8058299c66.png">
